packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts

```diff
@@ -202,8 +203,8 @@ export class MetricsExplorerService<
                         );
                     }
 
-                    const compareMetricDimensionGrain = dateRange
-                        ? getGrainForDateRange(dateRange)
+                    const compareMetricDimensionGrain = timeDimensionOverride
```

Comment:
we should keep the time dimension or the previous interval to get correct metric comparison

Reply:
correct. its done in the UI now so there is no need for that call